enterprise/com/src/main/java/org/neo4j/com/storecopy/TransactionCommittingResponseUnpacker.java

```diff
@@ -232,24 +232,45 @@ private void applyQueuedTransactions() throws IOException
                         // ensure that all pending changes are applied and flushed properly.
                         storeApplier.closeBatch();
                     }
+                    catch ( Throwable cause )
+                    {
+                        updateKernelHealth( cause );
+                        throw cause;
+                    }
                     finally
                     {
                         // Mark the applied transactions as closed. We must do this as a separate step after
                         // applying them, with a closeBatch() call in between, otherwise there might be
                         // threads waiting for transaction obligations to be fulfilled and since they are looking
                         // at last closed transaction id they might get notified to continue before all data
                         // has actually been flushed properly.
-                        transactionQueue.accept( batchCloser );
+                        if ( kernelHealth.isHealthy() )
+                        {
+                            transactionQueue.accept( batchCloser );
+                        }
```

Comment:
If it doesn't need to be in a `finally` clause, then this inner `try` clause can be removed.